wordpress-varnish-as-a-service.php

```diff
@@ -9,17 +9,86 @@
 GitHub Plugin URI: https://github.com/UCF/UCF-WordPress-Varnish-as-a-Service
 Description: A plugin for purging Varnish cache when content is published or edited. It works with HTTP purge and Admin Port purge. Works with Varnish 2 (PURGE) and Varnish 3 (BAN) versions. Based on WordPress Varnish and Plugin Varnish Purges.
 */
+if ( ! defined( 'WPINC' ) ) {
+    die;
+}
+
 class WPVarnish {
 	public $commenter;
+	public $is_multisite=false;
+	public $is_network_activated=false;
+	public $plugin_group='wp-varnish-aas';
+
 	/**
 	 * __construct
 	 * Set default values and options for plugin.
 	 * Should be backward compatible back to PHP4.
 	 * @since v2.0.0
 	 **/
-	function __construct() {
+	public function __construct() {
 		global $post;
+		global $is_multisite;
+		global $is_network_activated;
 
+		if ( !function_exists( 'is_plugin_active_for_network' ) ) {
```

Comment:
Something to keep in mind: I don't see anything that prevents this plugin from being loaded/instantiated on front-end loads, so this `require_once` would theoretically also occur on front-end requests, which probably isn't desirable for performance reasons.  It may be worth refactoring some of this plugin code to ensure this logic (and maybe other parts of the plugin) is only running on backend/admin requests as necessary.

Reply:
https://github.com/UCF/UCF-WordPress-Varnish-as-a-Service/pull/19/files#diff-354c4799a41a63b71f70afc294fb50dfR629